tests/integration_tests/saas/test_segment_task.py

```diff
@@ -131,3 +137,165 @@ def test_segment_saas_access_request_task(
     )
 
     assert filtered_results[f"{dataset_name}:segment_user"][0]["segment_id"]
+
+
+def _create_test_segment_email(base_email: str, timestamp: int) -> str:
+    at_index: int = base_email.find("@")
+    email = f"{base_email[0:at_index]}{timestamp}{base_email[at_index:]}"
+    return email
+
+
+def create_segment_test_data(
+    segment_secrets: Dict[str, str], segment_identity_email: str
+):
+    """Seeds a segment user and event"""
+    if not segment_identity_email:  # Don't run unnecessarily locally
+        return
+
+    faker = Faker()
+
+    ts = int(time.time())
+    email = _create_test_segment_email(segment_identity_email, ts)
+    first_name = faker.first_name()
+    last_name = faker.last_name()
+
+    # Create user
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Basic {segment_secrets['user_token']}",
+    }
+    body = {
+        "userId": email,
+        "traits": {
+            "subscriptionStatus": "active",
+            "address": {
+                "city": faker.city(),
+                "country": faker.country(),
+                "postalCode": faker.postcode(),
+                "state": "NY",
+            },
+            "age": random.randrange(18, 99),
+            "avatar": "",
+            "industry": "data",
+            "description": faker.job(),
+            "email": email,
+            "firstName": first_name,
+            "id": ts,
+            "lastName": last_name,
+            "name": f"{first_name} {last_name}",
+            "phone": faker.phone_number(),
+            "title": faker.prefix(),
+            "username": f"test_fidesops_user_{ts}",
+            "website": "www.example.com",
+        },
+    }
+    resp = requests.post(
+        f"https://{segment_secrets['api_domain']}identify", headers=headers, json=body
+    )
+    assert resp.status_code == 200
+
+    # Create event
+    body = {
+        "userId": email,
+        "type": "track",
+        "event": "User Registered",
+        "properties": {"plan": "Free", "accountType": faker.company()},
+        "context": {"ip": faker.ipv4()},
+    }
+
+    resp = requests.post(
+        f"https://{segment_secrets['api_domain']}track", headers=headers, json=body
+    )
+    assert resp.status_code == 200
+    return email
+
+
+@pytest.mark.integration_saas
+@pytest.mark.integration_segment
+def test_segment_saas_erasure_request_task(
+    db,
+    policy,
+    segment_connection_config,
+    segment_dataset_config,
+    segment_identity_email,
+    segment_secrets,
+) -> None:
+    """Full erasure request based on the Segment SaaS config"""
+    config.execution.MASKING_STRICT = False  # Allow GDPR Delete
+
+    # Create user for GDPR delete
+    erasure_email = create_segment_test_data(segment_secrets, segment_identity_email)
```

Comment:
Slight concerns in running this too frequently. To test a GDPR delete I am creating a user and then issuing the masking request. None of the segment masking requests I've made over the last several days have completed, as Segment completes them in batches. Does our account have limits on this?

Historically it looks like we were running these tests once a week or so.

Reply:
@eastandwestwind and @pattisdr according to segment's [API docs](https://segment.com/docs/privacy/user-deletion-and-suppression/), for the User Deletion and Suppression API: “[Contact Support](https://segment.com/help/contact/) if you need to process more than 100,000 users within a 30 day period.”

I think submitted a few every day would still be ok.

Reply:
thank you!

Reply:
yay, thanks!